solver/src/main/java/org/chocosolver/solver/constraints/nary/nvalue/PropNValue.java

```diff
@@ -0,0 +1,164 @@
+/*
+@author Arthur Godet <arth.godet@gmail.com>
+@since 13/03/2020
+*/
+package org.chocosolver.solver.constraints.nary.nvalue;
+
+import gnu.trove.list.array.TIntArrayList;
+import java.util.Random;
+import org.chocosolver.memory.IStateInt;
+import org.chocosolver.solver.constraints.Propagator;
+import org.chocosolver.solver.constraints.PropagatorPriority;
+import org.chocosolver.solver.exception.ContradictionException;
+import org.chocosolver.solver.variables.IntVar;
+import org.chocosolver.solver.variables.events.IntEventType;
+import org.chocosolver.solver.variables.events.PropagatorEventType;
+import org.chocosolver.util.ESat;
+import org.chocosolver.util.objects.setDataStructures.ISet;
+import org.chocosolver.util.objects.setDataStructures.SetFactory;
+import org.chocosolver.util.objects.setDataStructures.SetType;
+import org.chocosolver.util.tools.ArrayUtils;
+
+public class PropNValue extends Propagator<IntVar> {
+    private IntVar nValue;
+    private final int n;
+    private int[] concernedValues;
+    private IStateInt[] witness;
```

Comment:
There is no need for witnesses (or watch literal) to be backtrackable.
Indeed, if a value was considered as watched at a given time, it remains true on backtrack

Reply:
Indeed, I should use int[] instead of IStateInt[]